tnz/zti.py

```diff
@@ -1838,6 +1846,45 @@ def help_plugin(entry=entry):
 
         self.plugins = " ".join(plugins)
 
+    def __register_macros(self):
+        import importlib.util
+        import sys
+        import types
+
+        if not os.path.isdir(self.macros_dir):
+            return
+
+        for macro_file in os.listdir(self.macros_dir):
+            if not macro_file.endswith(".py"):
+                continue
+
+            if len(macro_file.split('.')) > 1:
+                continue
+
+            macro_name = macro_file.split('.')[0]
+
+            # Ignore macros which already exist
+            if f"do_{macro_name}" in self.get_names():
+                continue
+
+            macro_path = os.path.join(self.macros_dir, macro_file)
+
+            # Import the user macro as a module
+            macro_spec = importlib.util.spec_from_file_location(
+                f"module.{macro_name}", macro_path)
+            macro_module = importlib.util.module_from_spec(macro_spec)
+            sys.modules[f"module.{macro_name}"] = macro_module
+            macro_spec.loader.exec_module(macro_module)
+
+            # Find the function
+            if hasattr(macro_module, macro_name):
+                # Create a new bound method for the `Zti` class for this
+                # function
+                setattr(Zti, f"do_{macro_name}",
```

Comment:
Maybe it would make more sense if the function name in the .py file were do_{macro_name}... that way the name would be the same. And maybe make an optional help_{macro_name} so that help can be provided - for a "help {macro_name}" command.